app/src/main/java/com/io/moviO/adapter/MovieListAdapter.kt

```diff
@@ -0,0 +1,35 @@
+package com.io.moviO.adapter
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.io.moviO.Movie
+import com.io.moviO.databinding.ListItemBinding
+
+class MovieListAdapter(
+    private val listener: SelectListener,
```

Comment:
Usually most listeners start with On.., like OnClickListener, OnEditListener, OnFocusChangedListener so it will be nicer if you try to name your listeners in such way as well